### ISSUE TITLE
Make Xtext Sign and Deploy process more future proof

### ISSUE THREAD

**Issue by cdietrich** [eclipse/xtext#2065]
our current sign and deploy process is a big obscure mess nobody who cares anymore understands and that is full of tech. debt

- https://github.com/eclipse/xtext/issues/2057
- https://gitlab.eclipse.org/eclipsefdn/helpdesk/-/issues/1226
- https://github.com/xtext/publishing/issues/24
- https://github.com/eclipse/xtext/issues/1987

we use a stoneage gradle version and old centos and old java
that worries me a lot.

**Comment by cdietrich** [comment]
@kthoms @szarnekow @nprediger  this is something where i really need help

**Comment by kthoms** [comment]
When changing to centos-8 the build fails with
```
14:29:24  > Task :signArchivesXtextCore FAILED
14:29:24  :signArchivesXtextCore (Thread[Daemon worker,5,main]) completed. Took 0.022 secs.
14:29:24  
14:29:24  FAILURE: Build failed with an exception.
14:29:24  
14:29:24  * What went wrong:
14:29:24  Execution failed for task ':signArchivesXtextCore'.
14:29:24  > Unable to retrieve secret key from key ring file '/home/jenkins/.gnupg/secring.gpg' as it does not exist
```

**Comment by cdietrich** [comment]
@kthoms  yes this is where i was stuck

**Comment by kthoms** [comment]
We tried a gazillion of permutations. Next step is to disable the GPG magic in the publishing plugin and do this by shell scripting.
@nprediger and I have set a follow-up for June 22th

**Comment by cdietrich** [comment]
@kthoms @nprediger  i have found this one
https://github.com/xtext/publishing/blob/343c89fb465ba40f8cbfd0652752d47a845718a0/buildSrc/src/main/java/io/typefox/publishing/MavenPublishing.xtend#L331
here PublishingPlugin.SIGNING_PASSWORD
is set. but the gpgSign() thing seems to
`signing.password` and not signing.gnupg.passphrase (the keyname also needs to be 
`signing.gnupg.keyName`

**Comment by kthoms** [comment]
@nprediger and I are continuing today.

Seems that we have some progress now. 

```
14:45:33  Invoking gpg2 with arguments: [--local-user, D1AE0CFD, --pinentry-mode=loopback, --passphrase, ****, --no-tty, --batch, --detach-sign]
14:45:33  Starting process 'command 'gpg2''. Working directory: /home/jenkins/agent/workspace/releng/sign-and-deploy/publishing Command: gpg2 --local-user D1AE0CFD --pinentry-mode=loopback --passphrase **** --no-tty --batch --detach-sign
14:45:33  Successfully started process 'command 'gpg2''
14:45:33  Invoking gpg2 with arguments: [--local-user, D1AE0CFD, --pinentry-mode=loopback, --passphrase, ****, --no-tty, --batch, --detach-sign]
14:45:33  Starting process 'command 'gpg2''. Working directory: /home/jenkins/agent/workspace/releng/sign-and-deploy/publishing Command: gpg2 --local-user D1AE0CFD --pinentry-mode=loopback --passphrase **** --no-tty --batch --detach-sign
14:45:33  Successfully started process 'command 'gpg2''

...

14:46:35  Publishing to repository 'ossrh' (https://oss.sonatype.org/content/repositories/snapshots/)
14:46:35  Uploading maven-metadata.xml to /content/repositories/snapshots/org/eclipse/xtext/org.eclipse.xtext.builder.standalone/2.28.0-SNAPSHOT/maven-metadata.xml
14:46:35  Uploading org.eclipse.xtext.builder.standalone-2.28.0-20220622.124415-26.pom to /content/repositories/snapshots/org/eclipse/xtext/org.eclipse.xtext.builder.standalone/2.28.0-SNAPSHOT/org.eclipse.xtext.builder.standalone-2.28.0-20220622.124415-26.pom
14:46:35  Uploading org.eclipse.xtext.builder.standalone-2.28.0-20220622.124415-26-sources.jar to /content/repositories/snapshots/org/eclipse/xtext/org.eclipse.xtext.builder.standalone/2.28.0-SNAPSHOT/org.eclipse.xtext.builder.standalone-2.28.0-20220622.124415-26-sources.jar
14:46:35  Uploading org.eclipse.xtext.builder.standalone-2.28.0-20220622.124415-26-javadoc.jar to /content/repositories/snapshots/org/eclipse/xtext/org.eclipse.xtext.builder.standalone/2.28.0-SNAPSHOT/org.eclipse.xtext.builder.standalone-2.28.0-20220622.124415-26-javadoc.jar
14:46:35  Uploading org.eclipse.xtext.builder.standalone-2.28.0-20220622.124415-26-sources.jar.asc to /content/repositories/snapshots/org/eclipse/xtext/org.eclipse.xtext.builder.standalone/2.28.0-SNAPSHOT/org.eclipse.xtext.builder.standalone-2.28.0-20220622.124415-26-sources.jar.asc
14:46:35  Uploading org.eclipse.xtext.builder.standalone-2.28.0-20220622.124415-26.jar to /content/repositories/snapshots/org/eclipse/xtext/org.eclipse.xtext.builder.standalone/2.28.0-SNAPSHOT/org.eclipse.xtext.builder.standalone-2.28.0-20220622.124415-26.jar
14:46:35  Uploading org.eclipse.xtext.builder.standalone-2.28.0-20220622.124415-26-javadoc.jar.asc to /content/repositories/snapshots/org/eclipse/xtext/org.eclipse.xtext.builder.standalone/2.28.0-SNAPSHOT/org.eclipse.xtext.builder.standalone-2.28.0-20220622.124415-26-javadoc.jar.asc
14:46:35  Uploading org.eclipse.xtext.builder.standalone-2.28.0-20220622.124415-26.pom.asc to /content/repositories/snapshots/org/eclipse/xtext/org.eclipse.xtext.builder.standalone/2.28.0-SNAPSHOT/org.eclipse.xtext.builder.standalone-2.28.0-20220622.124415-26.pom.asc
14:46:35  Uploading org.eclipse.xtext.builder.standalone-2.28.0-20220622.124415-26.jar.asc to /content/repositories/snapshots/org/eclipse/xtext/org.eclipse.xtext.builder.standalone/2.28.0-SNAPSHOT/org.eclipse.xtext.builder.standalone-2.28.0-20220622.124415-26.jar.asc
14:46:35  Uploading maven-metadata.xml to /content/repositories/snapshots/org/eclipse/xtext/org.eclipse.xtext.builder.standalone/maven-metadata.xml
14:46:35  :publishOrgeclipsextextbuilderstandalonePublicationToOssrhRepository (Thread[Daemon worker,5,main]) completed. Took 1.699 secs.
```

**Comment by kthoms** [comment]
Based on the work from @cdietrich we have added the GPG credentials to the pipeline. This lead us to the publishing plugin since it then required the possibility to pass the passphrase as parameter. Further it then complained:
```
No value has been specified for property 'signatory.keyId'
```
We have seen that the property ''signing.gnupg.keyName' should be set to the value of 'signing.keyId'.

We did some adjustments on the publishing plugin to handle the additional properties. Further it seems that we have to use `useGpgCmd()` for the signing plugin. Without it did not work.
